drift_dev/lib/src/analysis/driver/state.dart

```diff
@@ -33,7 +33,9 @@ class FileState {
     final hasImports = discovery?.importDependencies.isNotEmpty == true;
     final hasQuery = analyzedElements.any((e) => e is DefinedSqlQuery);
 
-    return hasImports || hasQuery;
+    return hasImports ||
+        hasQuery ||
+        true; // TODO: This is required for other cases than just the above
```

Comment:
One of my tables was needed for the typedef function of a dynamic `$where` expression. This function was returning `false` which resulted in the `ModularAccessor` class not being created.

Reply:
If you have a scoped `$where`, that can only appear in a query so `hasQuery` should be true in that case right? Maybe this is another bug?

I'm happy to take a look at the original bug as well, but I think I don't fully understand how this could happen yet.